### PR TITLE
Join trip

### PR DESCRIPTION
**Pull request by FranzGithub**
This PR introduces functionality for Join Trip. It is not pretty. The button is always displayed. Nevermind if you are already part of the trip. One could check trip.passengers if the userid is already in there.

### CODE REVIEW

File: src/main/java/se/lth/base/server/rest/TripResource.java [hunk 1]
```diff
@@ -74,6 +75,14 @@ public Trip addTrip(Trip trip) throws URISyntaxException {
         trip.isCompleted());
   }
 
+  @Path("join/{tripid}/{userid}")
+  @POST
+  @PermitAll
```
Comment (jadlers) [inline src/main/java/se/lth/base/server/rest/TripResource.java]: Why permit all? Shouldn't this be restricted to users of the site?

File: src/main/resources/webassets/passenger/passengerSearch.html [hunk 2]
```diff
@@ -20,7 +20,7 @@ <h1> Sökresultat </h1>
               <td class="destinationAddress" ></td>
               <td class="driverId" ></td>
               <td>
-                <button class="btn btn-sm">Åk med</button>
+                <button class="btn btn-sm" type="button" onclick="joinTrip(this)">Åk med</button>
```
Comment (jadlers) [inline src/main/resources/webassets/passenger/passengerSearch.html]: Why `type="button"` on a `<button />`?

Comment (FranzGithub) [reply src/main/resources/webassets/passenger/passengerSearch.html]: A button without a type will act as a submit button. I prefer to define type=button on buttons that are not supposed to submit a form.

Comment (jadlers) [reply src/main/resources/webassets/passenger/passengerSearch.html]: I see, did not know that previously!

**Pull request merged**